source/MaterialXCore/Element.h

```diff
@@ -581,7 +581,10 @@ class Element : public enable_shared_from_this<Element>
     /// @param source The element from which content is copied.
     /// @param sourceUris If true, then source URIs from the given element
     ///    and its descendants are also copied.  Defaults to false.
-    void copyContentFrom(ConstElementPtr source, bool sourceUris = false);
+    /// @param skipDuplicateElements If true then skip copying any child Elements with
+    ///      if one with the same name already exists. Defaults to false.
+    void copyContentFrom(ConstElementPtr source, bool sourceUris = false,
```

Comment:
This approach will definitely work, but it's worth considering that we may need additional flags in the future, e.g. a "renameDuplicateElements" flag that assigns unique names to elements when conflicts occur.  From that perspective, it could be advantageous to encapsulate "sourceUris" and "skipDuplicateElements" in a CopyOptions class now, at the point that we're going from one flag to two.

From a functionality perspective, it seems logical for both "sourceUris" and "skipDuplicateElements" to be supported in importDocument as well, so the same CopyOptions class would work well in both situations.  It's more questionable whether "sourceUris" is meaningful in an XmlReadOptions class, which implies that we might want to omit the inheritance relationship between CopyOptions and XmlReadOptions, and consider them completely independent classes.

Reply:
For reuse, how about this proposed hierarchy:

ReadOptions { skipDuplicateElements=false; }
CopyOptions : public ReadOptions { sourceUris=true; }
XMLReadOptions : public ReadOptions { readXIncludes=true; } 

with this interface usage: 

Document::importLibrary( CopyOptions );
Element::copyContentsFrom( CopyOptions );
MaterialX::readFrom<...>( XMLReadOptions );

Reply:
It seems like it may be jumping the gun to create a Copy/Read/XmlRead class hierarchy now, without full knowledge of how these classes will evolve in the future.  I think the interface you describe above is exactly right, but I would recommend keeping CopyOptions and XmlReadOptions independent for now.  If there turns out to be a data/code sharing advantage in the future, we can easily add the inheritance relationship then.

Reply:
Sounds good. I'll make the changes.